### PR TITLE
Split CheepRepository into author and cheep repositories + unit tests + single select Query in GetCheepsFromUserTimelineAsync

### PR DESCRIPTION
**Pull request by ghost**
I separated our single "Cheeprepository" into "AuthorRepository" and "CheepRepository" with their respective methods.

Also added unit tests/integration tests for: 
-GetCheepsFromUserTimeline 
-Follow
-Unfollow

and tried to implement a single select Query in the GetCheepsFromUserTimelineAsync. Instead of two selects, I used the "Any()" statement which is the same as "EXISTS" in SQL.  And the "||" is an "OR" condition. It is more efficient, but I feel the original is more readable. 

The 

```
_dbContext.AuthorFollowers.Any(f => 
    f.FollowingId == cheep.AuthorId && 
    f.Follower.Name == author)"
    
    would be the same as:
```
    
  ```
  WHERE EXISTS (
    SELECT 1 
    FROM AuthorFollowers f
    WHERE f.FollowingId = cheep.AuthorId 
    AND f.Follower.Name = @author
)
```

### CODE REVIEW

**Review by AdrianVT — COMMENTED** [review]
Good overall, but need some smaller changes.

File: tests/unitTest/unitTest.csproj [hunk 1]
```diff
@@ -12,6 +12,7 @@
   <ItemGroup>
     <PackageReference Include="coverlet.collector" Version="6.0.0" />
     <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.8.0" />
+    <PackageReference Include="Moq" Version="4.20.72" />
```
Comment (cstiitu) [inline tests/unitTest/unitTest.csproj:15]: Remove use of Moq, we use inmemory database.

File: src/Chirp.Infrastructure/Chirp.Repositories/CheepRepository.cs [hunk 2]
```diff
@@ -70,70 +63,77 @@ orderby c.TimeStamp descending
         return await query.ToListAsync();
     }
 
-    public async Task <List<CheepDTO>> GetCheepsFromUserTimelineAsync(int page, string author)
+    // public async Task <List<CheepDTO>> GetCheepsFromUserTimelineAsync(int page, string author)
```
Comment (cstiitu) [inline src/Chirp.Infrastructure/Chirp.Repositories/CheepRepository.cs]: Remove unused method

File: tests/unitTest/UnitTest1.cs [hunk 3]
```diff
@@ -1,110 +1,140 @@
 namespace unitTest;
 
 using Chirp.Infrastructure.Chirp.Repositories;
+using Chirp.Infrastructure.Data;
+using Microsoft.EntityFrameworkCore;
 using Util;
+using Moq;
```
Comment (cstiitu) [inline tests/unitTest/UnitTest1.cs]: Remove moq

File: tests/unitTest/UnitTest1.cs [hunk 4]
```diff
@@ -1,110 +1,140 @@
...
 public class UnitTest1
 {
+    private readonly Mock<IAuthorRepository> _mockAuthorRepository;
+    private readonly IAuthorRepository _authorRepository;
+    private readonly ICheepRepository _cheepRepository;
+    private readonly ChirpDBContext _context;
+
+    public UnitTest1()
+    {
+        _mockAuthorRepository = new Mock<IAuthorRepository>();
```
Comment (cstiitu) [inline tests/unitTest/UnitTest1.cs]: Remove moq

**Review by cstiitu — CHANGES_REQUESTED** [review]
Remove Moq from testing and unused methods

**Review by Konradmeno — APPROVED** [review]

**Pull request merged**